webapp/params.go

```diff
@@ -28,9 +28,19 @@ const MaxCountMinValue = 1
 // SHARegex is a regex for SHA[0:10] slice of a git hash.
 var SHARegex = regexp.MustCompile("[0-9a-fA-F]{10,40}")
 
-// ParseSHAParam parses and validates the 'sha' param for the request.
-// It returns "latest" by default (and in error cases).
+// ParseSHAParam parses and validates the 'sha' param for the request,
+// cropping it to 10 chars. It returns "latest" by default. (and in error cases).
```

Comment:
Why is the cropping introduced? It looks like the longest ambiguous shas in wpt are 7 characters long (93d383f, d0b03af, e54fe8f) and it's fairly unlikely we will have a collision, but using full hashes everywhere internally and only accepting shorter ones as input seems like a good idea. But maybe this is necessary because 10-char hashes are parts of the filenames?

Reply:
I think Luke's aware of #49 , but many existing code doesn't take full SHAs so I assume this is a transient solution. At least this endpoint now takes a full SHA (despite chopping it internally), which is good.

Reply:
Note: This PR doesn't introduce the cropping, it bypasses it (since we want to use the full hash here).

See https://github.com/web-platform-tests/wpt.fyi/pull/46 for the temporary 'fix' as Hexcles describes.

Reply:
Oh... there was already a `runParam[:10]` before and this introduces a `ParseSHAParamFull`. Sorry, didn't finish reading before reacting :/